osu.Server.Queues.ScoreStatisticsProcessor/Processors/MedalAwarders/StarRatingMedalAwarder.cs

```diff
@@ -0,0 +1,154 @@
+    {
+        private BeatmapStore? beatmapStore;
+
+        public bool RunOnFailedScores => false;
+
+        public IEnumerable<Medal> Check(SoloScoreInfo score, UserStats userStats, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            return checkAsync(score, medals, conn, transaction).Result;
+        }
+
+        private async Task<IEnumerable<Medal>> checkAsync(SoloScoreInfo score, IEnumerable<Medal> medals, MySqlConnection conn, MySqlTransaction transaction)
+        {
+            List<Medal> earnedMedals = new List<Medal>();
+
+            Ruleset ruleset = LegacyRulesetHelper.GetRulesetFromLegacyId(score.RulesetID);
+            Mod[] mods = score.Mods.Select(m => m.ToMod(ruleset)).ToArray();
+
+            // Ensure the score isn't using any difficulty reducing mods
+            if (mods.Any(x => x is ModEasy || x is ModNoFail || x is ModHalfTime))
+                return earnedMedals;
+
+            try
+            {
+                beatmapStore ??= await BeatmapStore.CreateAsync(conn, transaction);
+
+                Beatmap? beatmap = await beatmapStore.GetBeatmapAsync(score.BeatmapID, conn, transaction);
+                if (beatmap == null)
+                    return earnedMedals;
+
+                // Get map star rating (including mods)
+                APIBeatmap apiBeatmap = beatmap.ToAPIBeatmap();
+                DifficultyAttributes? difficultyAttributes = await beatmapStore.GetDifficultyAttributesAsync(apiBeatmap, ruleset, mods, conn, transaction);
+                if (difficultyAttributes == null)
+                    return earnedMedals;
+
+                // Award pass medals
+                foreach (var medal in medals)
+                {
+                    if (checkMedalPass(score, medal, difficultyAttributes.StarRating))
+                        earnedMedals.Add(medal);
+                }
+
+                // Check for FC and award FC medals if necessary
+                if (score.MaxCombo == score.MaximumStatistics.Where(kvp => kvp.Key.AffectsCombo()).Sum(kvp => kvp.Value))
+                {
+                    foreach (var medal in medals)
+                    {
+                        if (checkMedalFc(score, medal, difficultyAttributes.StarRating))
+                            earnedMedals.Add(medal);
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                await Console.Error.WriteLineAsync($"{score.ID} failed in StarRatingMedalAwarder with: {ex}");
+            }
+
+            return earnedMedals;
+        }
+
+        private bool checkMedalPass(SoloScoreInfo score, Medal medal, double starRating)
+        {
+            // osu!standard 1-8*
+            if (checkMedalRange(55, medal.achievement_id, starRating))
+                return true;
+
+            // osu!taiko 1-8*
+            if (checkMedalRange(71, medal.achievement_id, starRating))
+                return true;
```

Comment:
https://osu.ppy.sh/beatmapsets/2626#taiko/19990 is explicitly disallowed for this medal range.